runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -100,6 +100,11 @@ Java_sun_misc_Unsafe_defineAnonymousClass(JNIEnv *env, jobject receiver, jclass
 
 	jsize length = env->GetArrayLength(bytecodes);
 
+	if(NULL != vm->sharedClassConfig){
+		/* if shared classes are enabled for an anonymous class, the class loader shared classes should be enabled to store the class in the cache */
```

Comment:
Better to mention isROMClassShareable() here in this comment that we are storing lamda classes to the shared cache.